queries.sql

```diff
@@ -8,3 +8,54 @@ SELECT name,escape_attempt FROM animals WHERE weight_kg>10.5;
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE name<>'Gabumon';
 SELECT * FROM animals WHERE weight_kg>=10.4 AND weight_kg<=17.3;
+
+-- First transaction block of codes
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+ROLLBACK;
+COMMIT;
+
+-- Second Transaction block of codes
+BEGIN;
+UPDATE animals SET species='digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species ='pokemon' WHERE species IS NULL;
+COMMIT;
+
+-- Third Transaction block of codes
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
+COMMIT;
+
+-- Fourth Transaction block of codes
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals SET weight_kg=(weight_kg*-1);
+ROLLBACK TO SP1;
+UPDATE animals SET weight_kg=(weight_kg*-1) WHERE weight_kg<0;
+COMMIT;
+
+-- how many animals are there?
+SELECT COUNT(*) AS total_animals FROM animals;
+
+-- how many animals have never tried to escape
+SELECT COUNT(*) AS no_escape FROM animals WHERE escape_attempt=0;
+
+-- what is the average weight on animals
+SELECT AVG(weight_kg) AS average_weight FROM animals;
+
```

Comment:
- [ ] The expected result for this query should be `15.55`, your output is `15.0111`. While there isnt much of a big difference kindly fix this error as with large data sets these differences can becom significant and cause you to have incorrect results.

Your output
![image](https://github.com/Adams-Ghub/vet-clinic-database/assets/81932478/1034b2de-0be4-4b55-b026-36f10728fb65)

#### Note please update the screenshot for this query as well upon making changes